flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryPartitionTest.java

```diff
@@ -179,166 +181,311 @@ public void testWrongNumericPartition() {
+                                "2023", StandardSQLTypeName.NUMERIC))
+                .isEqualTo("'2023'");
     }
 
     @Test
-    public void testDateTimeFormatMonthPartitionRestriction() {
-        TablePartitionInfo info =
-                new TablePartitionInfo(
-                        "event",
-                        BigQueryPartition.PartitionType.MONTH,
-                        StandardSQLTypeName.DATETIME);
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10 10:10:10"))
-                .isEqualTo("event BETWEEN '2010-10' AND '2010-11'");
+    public void testRetrievePartitionColumnType() {
+        StandardSQLTypeName retrieved =
+                BigQueryPartition.retrievePartitionColumnType(
+                        StorageClientFaker.SIMPLE_BQ_TABLE_SCHEMA, "ts");
+
+        Assertions.assertThat(retrieved).isEqualTo(StandardSQLTypeName.TIMESTAMP);
     }
 
     @Test
-    public void testDateTimeFormatYearPartitionRestriction() {
-        TablePartitionInfo info =
-                new TablePartitionInfo(
-                        "event",
-                        BigQueryPartition.PartitionType.YEAR,
-                        StandardSQLTypeName.DATETIME);
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10 10:10:10"))
-                .isEqualTo("event BETWEEN '2010' AND '2011'");
+    public void testCheckPartitionCompletedHour() {
+        PartitionIdWithInfo partitionWithInfo =
+                new PartitionIdWithInfo(
+                        "2023072801",
+                        new TablePartitionInfo(
+                                "temporal",
+                                BigQueryPartition.PartitionType.HOUR,
+                                StandardSQLTypeName.TIMESTAMP,
+                                // partition watermark is now
+                                Instant.now()));
+
+        PartitionIdWithInfoAndStatus partitionWithInfoAndStatus =
+                BigQueryPartition.checkPartitionCompleted(partitionWithInfo);
+
+        Assertions.assertThat(partitionWithInfoAndStatus.getStatus())
+                .isEqualTo(BigQueryPartition.PartitionStatus.COMPLETED);
     }
 
     @Test
-    public void testDateFormatDayPartitionRestriction() {
-        TablePartitionInfo info =
-                new TablePartitionInfo(
-                        "event", BigQueryPartition.PartitionType.DAY, StandardSQLTypeName.DATE);
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10"))
-                .isEqualTo("event BETWEEN '2010-10-10' AND '2010-10-11'");
+    public void testCheckPartitionCompletedDay() {
```

Comment:
Let's add edge cases near end of day, end of hour etc.

Reply:
Done